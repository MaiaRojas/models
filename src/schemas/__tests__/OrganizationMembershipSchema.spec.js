const mongoose = require('mongoose/browser');
const { OrganizationMembershipSchema, OrganizationSchema, UserSchema } = require('../')(mongoose);

describe('OrganizationMembershipSchema', () => {
  it('should fail validation when fields missing', () => {
    const doc = new mongoose.Document({}, OrganizationMembershipSchema);
    expect(doc.validateSync().errors).toMatchSnapshot();
  });

  it('should validate ...', (done) => {
    const organization = new mongoose.Document({}, OrganizationSchema);
    const user = new mongoose.Document({}, UserSchema);

    const employer = new mongoose.Document({
      user: user._id,
      organization: organization._id,
    }, OrganizationMembershipSchema);

    employer.validate((err) => {
      expect(err).toBe(null);
      done();
    });
  });
});
