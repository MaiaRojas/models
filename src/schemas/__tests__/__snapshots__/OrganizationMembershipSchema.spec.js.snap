// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrganizationMembershipSchema should fail validation when fields missing 1`] = `
Object {
  "organization": [ValidatorError: Path \`organization\` is required.],
  "phone": [ValidatorError: Path \`phone\` is required.],
  "position": [ValidatorError: Path \`position\` is required.],
  "user": [ValidatorError: Path \`user\` is required.],
}
`;
